client/src/App.js

```diff
@@ -122,42 +148,39 @@ class App extends Component {
     this.renderLoggedInRoutes = this.renderLoggedInRoutes.bind(this);
   }
 
-  componentDidMount() {
+  async componentWillMount() {
+    var role;
+    var profile;
     this.setState({
       fetchingAuth: true
-    }, async () => {
-      try {
-        const result = await makeCall({}, '/isLoggedIn', 'get')
+    }) 
+    try {
+      const result = await makeCall({}, '/isLoggedIn', 'get')
+      this.setState({
+        fetchingAuth: false
+      });
+      if (result && !result.error) {
```

Comment:
This is where the JWT is decoded and used to set the state
Regex lifted from stackoverflow, as is the parsing